packages/sirius-web/frontend/sirius-web-application/src/omnibox/usePublishLibraries.ts

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { gql, useMutation } from '@apollo/client';
+import { useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import {
+  GQLErrorPayload,
+  GQLPublishLibrariesMutationData,
+  GQLPublishLibrariesMutationVariables,
+  GQLPublishLibrariesPayload,
+  GQLSuccessPayload,
+  UsePublishLibrariesValue,
+} from './usePublishLibraries.types';
+
+const publishLibrariesMutation = gql`
+  mutation publishLibraries($input: PublishLibrariesInput!) {
+    publishLibraries(input: $input) {
+      __typename
+      ... on SuccessPayload {
+        messages {
+          level
+          body
+        }
+      }
+      ... on ErrorPayload {
+        messages {
+          level
+          body
+        }
+      }
+    }
+  }
+`;
+
+const isSuccessPayload = (payload: GQLPublishLibrariesPayload): payload is GQLSuccessPayload =>
+  payload.__typename === 'SuccessPayload';
+
+const isErrorPayload = (payload: GQLPublishLibrariesPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+export const usePublishLibraries = (): UsePublishLibrariesValue => {
+  const { addErrorMessage, addMessages } = useMultiToast();
+  const [performPublishLibraries, { loading, data }] = useMutation<
+    GQLPublishLibrariesMutationData,
+    GQLPublishLibrariesMutationVariables
+  >(publishLibrariesMutation, {
+    onCompleted: (data) => {
+      const { publishLibraries } = data;
+      if (isErrorPayload(publishLibraries)) {
+        addMessages(publishLibraries.messages);
+      }
+      if (isSuccessPayload(publishLibraries)) {
+        addMessages(publishLibraries.messages);
+      }
+    },
+    onError: () => {
+      addErrorMessage('An unexpected error has occurred, please refresh the page');
+    },
```

Comment:
If you are relying on this I suspect that there may be a lifecycle issue. Wait for the result before closing the dialog.

Reply:
I was working under the assumption that the publication process could take time, and thus it wasn't acceptable to keep the dialog open during the publication process. Based on your comment related to using a `CompletableFuture` on the data fetcher, I understand we can drop this assumption for the moment, and solve it later if we have performance issues related to publication.